File: 1980. Find Unique Binary String/main.py
# total sam

from typing import List

class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        n = len(nums)
        nums = set(nums)

        def dfs(str):
            if str in nums:
                return ""
            if len(str) == n:
                return str
            return dfs(str + "1") or dfs(str + "0")

        return dfs("")

File: 1980. Find Unique Binary String/main.go
package main

func findDifferentBinaryString(nums []string) string {
	n := len(nums)
	numsMap := make(map[string]bool)
	for _, n := range nums {
		numsMap[n] = true
	}
	var dfs func(str string) string
	dfs = func(str string) string {
		if numsMap[str] {
			return ""
		}
		if len(str) == n {
			return str
		}
		if res := dfs(str + "1"); res != "" {
			return res
		}
		return dfs(str + "0")
	}
	return dfs("")
}
